### PR TITLE
Nomad ACL bootstrap process

### LINKED ISSUE #147: Create a nomad bootstrap process for ACL
To make ACLs work in nomad you need to run through a bootstrap process. This should be automated with ansible.

### PR DESCRIPTION
**Pull request by pdmthorsrud**
Creating a draft PR that can run tests while I write documentation. Please do give feedback if you got any.
Closes #147

### CODE REVIEW

File: template/README.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ Any valid configuration added to these directories will be added to their respec
 #### Nomad ACLs
 Refer to [config variations](#config-variations) on how to toggle ACLs in nomad.
 
+When ACLs in Nomad are on the bootstrap management token will be available in vault under `secret/nomad/management-token` with the two key-value pairs `accessor-id` and `secret-id`. `secret-id` is the token itself. This can be accessed from inside the vagrant box with `vault kv get secret/nomad-bootstrap-token`, or by going to vault's UI on `localhost:8200` and signing in with the root token.
```
Comment (zhenik) [inline template/README.md:27]: Option with CLI, I would suggest run `vault` cli from localhost,
example:
```bash
vagrant ssh -c "VAULT_ADDR=http://127.0.0.1:8200 vault kv get secret/nomad-bootstrap-token"
```

But 
```
Error making API request.

URL: GET http://127.0.0.1:8200/v1/sys/internal/ui/mounts/secret/nomad-bootstrap-token
Code: 500. Errors:

* missing client token
```

Option vie UI works fine.

**Comment by zhenik** [comment]
## Suggestion 1
I would suggest in `template/test/Makefile` add extra vars `nomad_agent_acl_enable=false`

```
up:
	vagrant up --provision
	$(MAKE) clean
	# Option 1 how to turn on ACL
	ANSIBLE_ARGS='--extra-vars "consul_agent_acl_default_policy=deny" "nomad_agent_acl_enable=true"' vagrant up --provision
	#$(MAKE) clean
```

## And required fix
Terraform will fail without the token. 

```
TASK [Terraform] ***************************************************************
fatal: [default]: FAILED! => {
    "changed": false
}

MSG:

Terraform plan could not be created
STDOUT: Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

nomad_job.countdash: Refreshing state... [id=countdash]


STDERR: 
Error: error checking for job: Unexpected response code: 403 (Permission denied)
```

File: template/README.md (hunk 1 above)
Comment (pdmthorsrud) [reply template/README.md:27]: Ah, I've probably set VAULT_TOKEN in my environment. I'll add that to the doc, thanks.

**Comment by pdmthorsrud** [comment]
> ## Suggestion 1
> I would suggest in `template/test/Makefile` add extra vars `nomad_agent_acl_enable=false`
> 
> ```
> up:
> 	vagrant up --provision
> 	$(MAKE) clean
> 	# Option 1 how to turn on ACL
> 	ANSIBLE_ARGS='--extra-vars "consul_agent_acl_default_policy=deny" "nomad_agent_acl_enable=true"' vagrant up --provision
> 	#$(MAKE) clean
> ```
> 
> ## And required fix
> Terraform will fail without the token.
> 
> ```
> TASK [Terraform] ***************************************************************
> fatal: [default]: FAILED! => {
>     "changed": false
> }
> 
> MSG:
> 
> Terraform plan could not be created
> STDOUT: Refreshing Terraform state in-memory prior to plan...
> The refreshed state will be used to calculate this plan, but will not be
> persisted to local or remote state storage.
> 
> nomad_job.countdash: Refreshing state... [id=countdash]
> 
> 
> STDERR: 
> Error: error checking for job: Unexpected response code: 403 (Permission denied)
> ```

This is important, yeah. I think this should be its own PR, because this is easily fixed when the nomad secret backend is enabled and roles have been created in vault.

File: template/README.md (hunk 1 above)
Comment (pdmthorsrud) [reply template/README.md]: Fixed

Comment (zhenik) [reply template/README.md]: Pay attention that I have also marked `VAULT_ADDR=http://127.0.0.1:8200`, because by default cli use `https` not `http`

Comment (pdmthorsrud) [reply template/README.md]: Crap, yep, you're right.

Comment (pdmthorsrud) [reply template/README.md]: There we go (I hope, haha)!

**Pull request merged**